### PR TITLE
fix(updater): installation on macos #591

### PR DESCRIPTION
**Pull request by pashokitsme**
Using flate2 to decompressing tar.gz

### CODE REVIEW

File: plugins/updater/Cargo.toml [hunk 1]
```diff
@@ -32,6 +32,10 @@ tar = "0.4"
 [target."cfg(target_os = \"windows\")".dependencies]
 zip = { version = "0.6", default-features = false }
 
+[target."cfg(target_os = \"macos\")".dependencies]
+flate2 = "1.0.27"
```
Comment (amrbashir) [inline plugins/updater/Cargo.toml:36]: the `zip` crate has `flate2` feature, could we use that instead of adding a new dependencies entry?

**Review by amrbashir — CHANGES_REQUESTED** [review]
cc @lucasfernog since I can't test this

File: plugins/updater/Cargo.toml (hunk 1 above)
Comment (pashokitsme) [reply plugins/updater/Cargo.toml:36]: Sure
I didn't notice zip has such feature

**Comment by FabianLars** [comment]
Since this PR was just mentioned in the other issue? Isn't it enough to literally just enable the feature in the zip crate? at a glance it looks like that's the only difference between the code in the plugin and the one in tauri v1 🤔

**Comment by pashokitsme** [comment]
@FabianLars 
Updater for macos uses `.tar.gz` archives, so they unpacks with crate `tar` which only reads archives, not decompresses. In Tauri v1 [updater](https://github.com/tauri-apps/tauri/blob/1.3/core/tauri/src/updater/core.rs#L846) uses [ArchiveFormat & ArchiveReader](https://github.com/tauri-apps/tauri/blob/1.3/core/tauri/src/api/file/extract.rs#L18) which handles zip and tar archives and requires `flate2` under the hood

**Comment by FabianLars** [comment]
Ah lol right, I forgot about our custom implementation, sorry!

**Review by lucasfernog — APPROVED** [review]

**Pull request merged**